mongo/change_stream.go

```diff
@@ -451,7 +451,7 @@ func (cs *changeStream) Decode(out interface{}) error {
 		return err
 	}
 
-	return bson.UnmarshalWithRegistry(cs.coll.registry, br, out)
+	return bson.UnmarshalWithRegistry(cs.db.registry, br, out)
```

Comment:
This is too restrictive, I think what we actually want is this:
```suggestion
	return bson.UnmarshalWithRegistry(cs.registry, br, out)
```